onnxscript/values.py

```diff
@@ -104,49 +176,54 @@ def __init__(
         self.opset = opset
         self.opname = opname
         self.opschema = opschema
+        self._param_schemas: Optional[tuple[ParamSchema, ...]] = None
+
+    def __call__(self, *args, **kwargs):
+        # FIXME(after #225): Move import to the top of the file.
+        from onnxscript import evaluator  # pylint: disable=import-outside-toplevel
+
+        return evaluator.default().eval(self, args, kwargs)
 
     def is_single_op(self) -> bool:
         return isinstance(self.opname, str)
 
     def get_schema(self) -> onnx.defs.OpSchema:
+        """Returns the ONNX OpSchema for this op."""
         if self.opschema:
             return self.opschema
         return self.opset[self.opname]
 
     def has_schema(self) -> bool:
+        """Returns True if this op has an OpSchema."""
         return self.opschema is not None
 
-    def adapt_kwargs(self, kwargs):
-        """Replaces function-valued attribute-values by their GraphProto representation."""
-        closure: dict[str, Any] = {}
-        for k, v in kwargs.items():
-            if isinstance(v, OnnxClosure):
-                kwargs[k] = v.function_ir.to_graph_proto()
-                for pyvar, onnxvar in v.function_ir.outer_scope_variables:
-                    closure[onnxvar.value] = v.frame.f_locals[pyvar]
-            elif callable(v):
-                raise ValueError(
-                    f"Error: function-valued attribute {v.__name__!r} has no graph_proto"
-                    "attribute. Did you forget to decorate it with @graph?"
-                )
-        return kwargs, closure
-
-    def _convert_kwargs_to_numpy(self, kwargs: dict[str, Any]) -> dict[str, Any]:
-        new_kwargs = {}
-        for k, v in kwargs.items():
-            new_kwargs[k] = v
-            if isinstance(v, tensor.Tensor):
-                new_kwargs[k] = v.value
-
-        return new_kwargs
-
-    def __call__(self, *args, **kwargs):
-        # FIXME(after #225): Move import to the top of the file.
-        from onnxscript import evaluator  # pylint: disable=import-outside-toplevel
-
-        return evaluator.default().eval(
-            self.opschema, args, self._convert_kwargs_to_numpy(kwargs)
-        )
+    def param_schemas(self) -> tuple[ParamSchema, ...]:
```

Comment:
May be introducing a type-alias `Schema = tuple[ParamSchema, ...]` would be helpful.

Reply:
Would it be confused with OpSchema? I was thinking if there's a more accurate name... Any suggestions?

Reply:
I was thinking of that too. One possibility is a name like `IRSchema` since this is really part of the IR we are using (which has counterparts to many things in ONNX).